Node/src/TeamsChatConnector.ts

```diff
@@ -171,17 +184,27 @@ export class TeamsChatConnector extends builder.ChatConnector {
         // Add intent message
         if (intentMessage.type == 'message') {
           var atMentions = [];
+          var botId: string = null;
+          if (re.address && re.address.bot && re.address.bot.id) {
+            botId = re.address.bot.id;
+          }
+
           intentMessage.intentText = intentMessage.text;
           for(var ent of intentMessage.entities) {
-            if (ent.type == 'mention') {
+            if (ent.type == 'mention' && ent.mentioned && botId && ent.mentioned.id == botId) {
               atMentions.push(ent.text);
             }
           }
+
           if (atMentions.length != 0) {
             for(var mention of atMentions) {
               intentMessage.intentText = intentMessage.intentText.replace(new RegExp(mention, 'g'), '');
```

Comment:
We can fold this into the loop above, instead of building atMentions array first then iterating over atMentions.